[Docs] For CCS and CCR local cluster determines priviliges of API key (#98205)
elastic/elasticsearch: x-pack/docs/en/security/authentication/remote-clusters-privileges.asciidoc

the `read_ccr` cluster privilege, and `monitor` and `read` privileges on the
leader index.

> NOTE: If requests will be authenticated with an <<security-api-create-api-key, API key>>, the API key

Author: This (and an identical callout for CCS below) is the only real change.

> NOTE: If requests will be authenticated with an <<security-api-create-api-key, API key>>, the API key

Reviewer: ```suggestion
NOTE: If requests are authenticated with an <<security-api-create-api-key, API key>>, the API key
```
Use present tense where possible.

Author: 👍 I'll fix the other call out as well (the one about run-as)

requires the above privileges on the **local** cluster, instead of the remote.

NOTE: If requests will be issued <<run-as-privilege,on behalf of other users>>,
then the authenticating user must have the `run_as` privilege on the remote

> cluster.

Reviewer 2: I wonder whether we should mention that the API Key callout applies to the `run_as` privilege here as well. Or maybe it could get too complicated since the target user of `run_as` needs to be on the *remote* cluster?

Author: Hmm that's a good question. I think it's too complicated to be honest -- the flow at that point would be:

Define role on the remote
Unless it's an API key
Oh wait, if it's an API key using run-as then ... 

My preference would be to stick with the current formulation. Let me know if you feel strongly -- happy to try another tweak in a follow up.

On the remote cluster, the {ccs} role requires the `read` and
`read_cross_cluster` privileges for the target indices.

> NOTE: If requests will be authenticated with an <<security-api-create-api-key, API key>>, the API key

Reviewer: ```suggestion
NOTE: If requests are authenticated with an <<security-api-create-api-key, API key>>, the API key
```
